src/Hardmon.cpp

```diff
@@ -0,0 +1,122 @@
+#include <Hardmon.hpp>
+
+namespace VCU {
+
+Hardmon::Hardmon(reqGPIO gpios, IO::CAN& ptCAN) : gpios(gpios), powertrainCAN(ptCAN) {
+}
+
+CO_OBJ_T* Hardmon::getObjectDictionary() {
+    return &objectDictionary[0];
+}
+
+uint8_t Hardmon::getNumElements() {
+    return OBJECT_DICTIONARY_SIZE;
+}
+
+uint8_t Hardmon::getNodeID() {
+    return NODE_ID;
+}
+
+void Hardmon::handlePowertrainCanMessage(IO::CANMessage& message) {
+    switch (message.getId()) {
+    case DEV::PowertrainCAN::HIB_MESSAGE_ID:
+        forwardEnable = powertrainCAN.parseHIBForwardEnable(message);
+        break;
+    case DEV::PowertrainCAN::UC_SELF_TEST_MESSAGE_ID:
+        powertrainCAN.sendHardmonSelfTestResponse();
+        break;
+    default:
+        //we don't care about this message lol
+        break;
+    }
+}
+
+EVT::core::types::FixedQueue<POWERTRAIN_QUEUE_SIZE, IO::CANMessage>* Hardmon::getPowertrainQueue() {
+    return &powertrainCAN.queue;
```

Comment:
If you create the queue in main and pass it in, you don't have to expose this pointer like this